.github/workflows/test-webservice.yml

```diff
@@ -16,37 +16,43 @@ defaults:
 
 jobs:
   macos:
-    name: macOS ${{ matrix.configuration }} Test TestWebService
+    name: macOS ${{ matrix.configuration }} test TestWebService
     runs-on: macos-latest
     strategy:
-      fail-fast: false
       matrix:
         configuration: [debug, release]
     steps:
     - uses: actions/checkout@v2
     - uses: maxim-lobanov/setup-xcode@v1.1
       with:
         xcode-version: latest
+    - uses: actions/cache@v2
+      with:
+        path: .build
+        key: ${{ runner.os }}-spm-cachev2-${{ hashFiles('**/Package.resolved') }}
```

Comment:
The TestWebService Action uses the same caching key pattern as the "regular" action. If the hash function would evaluate to the same result (they don't) they would store conflicting .build directories.

The hash function don't evaluate to the same value though as of the changed working directory, thus the `**/Package.resolved` will only find `TestWebService/Package.resolved` (I assume).
Though I would argue to make that difference explicit using a different key pattern.

Another note. As `**/Package.resolved` is still used in the "default" action, the hash will consider both the root `Package.resolved` and the `TestWebService/Package.resolved`, which is a bit weird. Maybe its worth removing the `**` matching. I can't find a case where it makes sense, wonder why they put that into the example over at `action/cache`.

The test-webservice action never touches the root `.build` folder right?

Reply:
That's a good point, I will update the PR to only include the correct `Package.resolved` file 👍